src/FixerConfiguration/FixerConfigurationResolver.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\FixerConfiguration;
+
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+final class FixerConfigurationResolver implements FixerConfigurationResolverInterface
+{
+    /**
+     * @var array<string, FixerOptionInterface>
+     */
+    private $options = array();
```

Comment:
The keys are just used internally to easily know whether an option is defined using `array_key_exists()`. Not sure how using `SplObjectStorage` would be better, but why not?